internal/controller/promotionstrategy_controller.go

```diff
@@ -201,126 +205,59 @@ func (r *PromotionStrategyReconciler) calculateStatus(ctx context.Context, ps *p
-			var csList promoterv1alpha1.CommitStatusList
-			// Find all the commit status that match the configured proposed commit status name and the sha of the proposed hydrated commit
-			err := r.List(ctx, &csList, &client.ListOptions{
-				LabelSelector: labels.SelectorFromSet(map[string]string{
-					"promoter.argoproj.io/commit-status": utils.KubeSafeLabel(ctx, status.Key),
-				}),
-				FieldSelector: fields.SelectorFromSet(map[string]string{
-					".spec.sha": pc.Status.Proposed.Hydrated.Sha,
-				}),
-			})
-			if err != nil {
-				return err
-			}
-
-			// We don't want to capture any of the copied commits statuses that are used for GitHub/Provider UI experience only.
-			csListSlice := []promoterv1alpha1.CommitStatus{}
-			for _, item := range csList.Items {
-				if item.Labels["promoter.argoproj.io/commit-status-copy"] != "true" {
-					csListSlice = append(csListSlice, item)
-				}
-			}
-
-			if len(csListSlice) == 1 {
-				allProposdedCSList = append(allProposdedCSList, csListSlice[0])
-			} else if len(csListSlice) > 1 {
-				// TODO: decided how to bubble up errors
-				ps.Status.Environments[i].Proposed.CommitStatus.State = "to-many-matching-sha"
-				ps.Status.Environments[i].Proposed.CommitStatus.Sha = "to-many-matching-sha"
-				r.Recorder.Event(ps, "Warning", "ToManyMatchingSha", "There are to many matching sha's for the proposed commit status")
-			} else if len(csListSlice) == 0 {
-				ps.Status.Environments[i].Proposed.CommitStatus.State = "no-commit-status-found"
-				ps.Status.Environments[i].Proposed.CommitStatus.Sha = "no-commit-status-found"
-				r.Recorder.Event(ps, "Warning", "NoCommitStatusFound", "No commit status found for the proposed commit")
-			}
-
-		}
-		if len(proposedCommitStatusList) == 0 && i >= 0 {
-			// If there is no configured proposed CommitStatuses, we assume success
-			ps.Status.Environments[i].Proposed.CommitStatus.State = string(promoterv1alpha1.CommitStatusSuccess)
-			ps.Status.Environments[i].Proposed.CommitStatus.Sha = pc.Status.Active.Hydrated.Sha
-		} else if i >= 0 {
-			// Loop through allActiveCSList and bubble up success if all are successful
-			ps.Status.Environments[i].Proposed.CommitStatus.State = string(promoterv1alpha1.CommitStatusSuccess)
-			for _, cs := range allActiveCSList {
-				if cs.Status.State != promoterv1alpha1.CommitStatusSuccess {
-					ps.Status.Environments[i].Proposed.CommitStatus.State = string(cs.Spec.State)
-					ps.Status.Environments[i].Proposed.CommitStatus.Sha = cs.Spec.Sha
+		proposedCommitStatusCounts := len(append(environment.ProposedCommitStatuses, ps.Spec.ProposedCommitStatuses...))
+		if proposedCommitStatusCounts > 0 && len(pcMap[environment.Branch].Status.Proposed.CommitStatuses) == proposedCommitStatusCounts {
+			// We have configured proposed commits and our count of proposed commits from promotion strategy matches the count of proposed commit resource.
+			for _, status := range pcMap[environment.Branch].Status.Proposed.CommitStatuses {
```

Comment:
I'm following your race condition remark from earlier now. It's possible the PC controller is reporting the correct _number_ of statuses without reporting the actually-required statuses. I'm not sure I'm super bothered by that right now.